### PR TITLE
Add date filters

### PR DESCRIPTION
**Pull request by burnNturn**
I added date pickers to filter the date also.  Everything works great with the server and database logic, but I couldn't get the "end_date" datepicker to default to today (tried using jquery, seems like it should be working), I think I have a way around this, but it seems a little sloppy.  

I also wanted to use jquery(?) to only render the table after a filter, which it is doing, but something is going wrong in my logic, it seems to be using @customers.all to fill the table (before I added the jquery it was filtering by date correctly, so it isn't that).

Just a heads up, I have two byebugs in there right now (one in customers_controller.index and one in the customers model) and I have a debugger in my index.js.erb.

Can you take a look at these, and see what I am doing wrong, or at least lead me into the right direction?

### CODE REVIEW

File: app/controllers/customers_controller.rb [hunk 1]
```diff
@@ -5,13 +5,19 @@ class CustomersController < ApplicationController
...
+                      params[:end_date].empty? && params[:age].empty? &&
+                      params[:gender].empty? && params[:income].empty? &&
+                      params[:race].empty? && params[:game].empty?)
                     Customer.all
                   else
                     Customer.search(params)
                   end
+    respond_to do |format|
+        format.js
+        format.html
```
Comment (Jabronious) [inline app/controllers/customers_controller.rb]: do you need the format.html?

File: app/views/customers/index.html.erb [hunk 2]
```diff
@@ -2,9 +2,21 @@
 
 <h1>Customers</h1>
 <div class="search-menu">
-  <%= form_tag customers_path, method: :get do %>
+  <%= form_tag customers_path, method: :get, remote: true do %>
     <div>
-      <%= select_tag :game, options_from_collection_for_select(Customer::game_names, :first, :last), 
+      <script>
+        $( function() {
+          $( "#start_date" ).datepicker();
```
Comment (Jabronious) [inline app/views/customers/index.html.erb]: you should be able to do ("#start_date, #end_date").datepicker();

Comment (Jabronious) [reply app/views/customers/index.html.erb]: Also, inside .datepicker([PUT_DEFAULT_STUFF]); ... check documentation on what you can put in there though not 100% sure names and such

**Review by Jabronious — CHANGES_REQUESTED** [review]

File: app/controllers/customers_controller.rb (hunk 1 above)
Comment (burnNturn) [reply app/controllers/customers_controller.rb]: Seems that I do, I believe that when the page initially loads, it is an html request.  I get an "UnknownFormat" error on the initial page load if I remove it.

**Comment by burnNturn** [comment]
All issues resolved.

**Pull request merged**